core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -175,6 +178,28 @@ abstract class AbstractFetcherThread(name: String,
     }
   }
 
+  private def addFailedPartition(topicPartition: TopicPartition): Unit = {
+    markPartitionFailed(topicPartition)
+    removePartitions(Set(topicPartition))
+  }
+
+  private def doTruncate(topicPartition: TopicPartition, truncationState: OffsetTruncationState): Boolean = {
+    try {
+      truncate(topicPartition, truncationState)
+      true
+    }
+    catch {
+      case e: KafkaStorageException =>
+        info(s"Failed to truncate $topicPartition", e)
```

Comment:
I wonder if it's worth adding the truncation offset to this log line and the one below.

Reply:
I have added the truncation offset to make the log more informative.